src/nbl/ext/CentralLimitBoxBlur/CBlurPerformer.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/ext/CentralLimitBoxBlur/CBlurPerformer.h"
+
+using namespace nbl;
+using namespace nbl::asset;
+using namespace nbl::video;
+using namespace ext::CentralLimitBoxBlur;
+
+CBlurPerformer::CBlurPerformer(video::IVideoDriver* driver, uint32_t maxDimensionSize, bool useHalfStorage)
+    : m_maxBlurLen(maxDimensionSize), m_halfFloatStorage(useHalfStorage)
+{
+    static IGPUDescriptorSetLayout::SBinding bnd[] =
+    {
+        {
+            0u,
+            EDT_STORAGE_BUFFER,
+            1u,
+            ISpecializedShader::ESS_COMPUTE,
+            nullptr
+        },
+        {
+            1u,
+            EDT_STORAGE_BUFFER,
+            1u,
+            ISpecializedShader::ESS_COMPUTE,
+            nullptr
+        },
+    };
+
+    m_dsLayout = driver->createGPUDescriptorSetLayout(bnd, bnd + sizeof(bnd) / sizeof(IGPUDescriptorSetLayout::SBinding));
+
+    auto pcRange = getDefaultPushConstantRanges();
+    m_pplnLayout = driver->createGPUPipelineLayout(pcRange.begin(), pcRange.end(), core::smart_refctd_ptr(m_dsLayout));
+
```

Comment:
I think we should create a `static` `create` function instead of a constructor?

Reply:
yes, agreed 100%